tools/keypoints2coco_without_mmdet.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import json
+from tqdm import tqdm
+def main():
+    """Visualize the demo images.
+
+    pose_keypoints equire the json_file containing boxes.
+    """
+    parser = ArgumentParser()
+    parser.add_argument('pose_config', help='Config file for detection')
+    parser.add_argument('pose_checkpoint', help='Checkpoint file')
+    parser.add_argument('--img-root', type=str, default='', help='Image root')
+    parser.add_argument(
+        '--json-file',
+        type=str,
+        default='',
+        help='Json file containing image person bboxes in COCO format.')
+    parser.add_argument(
+        '--out-json-file',
+        type=str,
+        default='',
+        help='Output json containing bboxes from Json file and pseudolabeled keypoints')
+    parser.add_argument(
+        '--show',
+        action='store_true',
+        default=False,
+        help='whether to show img')
+    parser.add_argument(
+        '--device', default='cuda:0', help='Device used for inference')
+    parser.add_argument(
+        '--kpt-thr', type=float, default=0.3, help='Keypoint score threshold')
+
+    args = parser.parse_args()
+
+    assert args.show or (args.out_img_root != '')
+
+    coco = COCO(args.json_file)
+    # build the pose model from a config file and a checkpoint file
+    pose_model = init_pose_model(
+        args.pose_config, args.pose_checkpoint, device=args.device.lower())
+
+    dataset = pose_model.cfg.data['test']['type']
+
+    img_keys = list(coco.imgs.keys())
+
+    # optional
+    return_heatmap = False
+
+    # e.g. use ('backbone', ) to return backbone feature
+    output_layer_names = None
+
+    categories = [{'id': 1, 'name': 'person'}]
+    img_anno_dict = {'images': [], 'annotations':[], 'categories':categories}
+
+    # process each image
+    ann_uniq_id = int(0)
+    for idx, i in enumerate(tqdm(range(len(img_keys)))):
+        # get bounding box annotations
+        image_id = img_keys[i]
+        image = coco.loadImgs(image_id)[0]
+        image_name = os.path.join(args.img_root, image['file_name'])
+        width, height = imagesize.get(image_name)
+        ann_ids = coco.getAnnIds(image_id)
+
+        # make person bounding boxes
+        person_results = []
+        for ann_id in ann_ids:
+            person = {}
+            ann = coco.anns[ann_id]
+            # bbox format is 'xywh'
+            person['bbox'] = ann['bbox']
+            person_results.append(person)
+
+        pose_results, returned_outputs = inference_top_down_pose_model(
+            pose_model,
+            image_name,
+            person_results,
+            bbox_thr=None,
+            format='xywh',
+            dataset=dataset,
+            return_heatmap=return_heatmap,
+            outputs=output_layer_names)
+    
+        #add output of model and bboxes to dict
+        for indx, i in enumerate(pose_results):
+            pose_results[indx]['keypoints'][::, 2] = int(2)
+            top = int(pose_results[indx]['bbox'][0])
```

Comment:
Please double check this. Does not `pose_results[indx]['bbox'][0]` mean left?

Reply:
Yes, my wrong but 
next lines
```
bbox_width = int(pose_results[indx]['bbox'][2] - pose_results[indx]['bbox'][0])
            bbox_height = int(pose_results[indx]['bbox'][3] - pose_results[indx]['bbox'][1])
            bbox = [top, left, bbox_width, bbox_height]
            area = round((bbox_width*bbox_height), 0)
```
must calculate right
I guess will be enough to rename variable to x, y?

Reply:
Yes. x, y, w, h is good enough.

Reply:
Fixed